src/ast/mod.rs

```diff
@@ -893,26 +1030,44 @@ where
 
 impl<Head, Leaf, Ann> GenericFact<Head, Leaf, Ann>
 where
-    Ann: Clone,
+    Ann: Clone + Default,
     Head: Clone + Display,
     Leaf: Clone + PartialEq + Eq + Display + Hash,
 {
-    pub(crate) fn map_exprs<Head2, Leaf2>(
-        &self,
-        f: &mut impl FnMut(&GenericExpr<Head, Leaf, Ann>) -> GenericExpr<Head2, Leaf2, Ann>,
-    ) -> GenericFact<Head2, Leaf2, Ann> {
+    pub(crate) fn map_exprs(
```

Comment:
This now seems to be a very different function than the original declaration. Should we keep the original method, which also seems useful? The proposed `map_exprs` seems more like `normalize` in the sense it does a bottom up normalizing pass.

Reply:
added old map_exprs, re-named mine to visit